core/src/main/java/org/apache/iceberg/GenericDataFile.java

```diff
@@ -115,14 +116,6 @@ public GenericDataFile(org.apache.avro.Schema avroSchema) {
     this.partitionType = EMPTY_PARTITION_DATA.getPartitionType();
     this.recordCount = recordCount;
     this.fileSizeInBytes = fileSizeInBytes;
-    this.fileOrdinal = null;
```

Comment:
I don't think this change is necessary, so it would be better to leave these as they are.

Reply:
These fields are already initialized to null, so I did some eager cleanup to remove these redundant assignments.